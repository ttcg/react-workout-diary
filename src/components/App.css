html {
  position: relative;
  min-height: 100%;
}

.content {
  margin-top: 10px;
  padding-bottom: 2.5rem;
}

.footer {
  position: absolute;
  bottom: 0;
  width: 100%;
  height: 2.5rem;
  line-height: 2.5rem;
  background-color: #f5f5f5;
  border-top: 1px solid #bcc5ce;
}

.form-signin {
  width: 100%;
  max-width: 330px;
  padding: 15px;
  margin: 0 auto;
}