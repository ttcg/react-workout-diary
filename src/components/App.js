import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import './App.css';
import es6Promise from 'es6-promise';
import es6ObjectAssign from 'es6-object-assign';
import Header from './common/header';
import Footer from './common/footer';
import Home from './home.js';
import WorkoutListReact from '../components/workouts/index';
import WorkoutListPage from '../pages/workoutListPage';
import workoutListApiPage from '../pages/workoutListApiPage';

es6Promise.polyfill();
es6ObjectAssign.polyfill();

class App extends Component {
  render() {
    return (      
        <BrowserRouter>
          <React.Fragment>
            <Header />
            <header className="App-header">

              <Switch>
                <Route path='(/|/home)' exact component={Home} />
                <Route path='/workouts' exact component={WorkoutListPage} />
                <Route path='/workoutsapi' exact component={workoutListApiPage} />
                <Route path='/workoutsreact' exact component={WorkoutListReact} />
              </Switch>

            </header>
            <Footer />
          </React.Fragment>
        </BrowserRouter>
    );
  }
}

export default App;
