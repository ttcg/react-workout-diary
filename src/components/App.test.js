import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';

it('renders without crashing', () => {
  const component = shallow(<App />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
