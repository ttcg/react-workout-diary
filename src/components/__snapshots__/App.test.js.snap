// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<BrowserRouter>
  <Header />
  <header
    className="App-header"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="(/|/home)"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/workouts"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/workoutsapi"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/workoutsreact"
      />
    </Switch>
  </header>
  <Footer />
</BrowserRouter>
`;
